File: app/src/main/java/edu/application/ui/fragments/settings/ColorSchemeFragment.java
package edu.application.ui.fragments.settings;

import static android.content.Context.MODE_PRIVATE;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.Navigation;

import edu.application.R;
import edu.application.databinding.FragmentColorsSchemeBinding;
import edu.application.ui.adapters.TextFormatter;

public class ColorSchemeFragment extends Fragment {

    private FragmentColorsSchemeBinding binding;
    private SharedPreferences sharedPreferences;
    private SharedPreferences.Editor editor;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {

        binding = FragmentColorsSchemeBinding.inflate(inflater, container, false);

        sharedPreferences = requireActivity().getPreferences(MODE_PRIVATE);
        editor = sharedPreferences.edit();

        TextFormatter.setColor(binding.customBtn);
        TextFormatter.setBorder(binding.customBtn);

        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        binding.lightBtn.setOnClickListener(v -> {
            editor.remove("lesson_background_color");
            editor.remove("lesson_text_color");
            editor.apply();
            Navigation.findNavController(requireActivity(), R.id.nav_host_fragment)
                    .navigate(R.id.action_colorSchemeFragment_to_colorSchemeFragment);
        });

        binding.customBtn.setOnClickListener(v ->
                Navigation.findNavController(requireActivity(), R.id.nav_host_fragment)
                        .navigate(R.id.action_colorSchemeFragment_to_colorSchemeCustomFragment));
    }
}

File: app/src/main/kotlin/edu/application/ui/fragments/navigation/account/settings/ColorSchemeFragment.kt
package edu.application.ui.fragments.navigation.account.settings

import android.content.SharedPreferences
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.navigation.Navigation.findNavController
import edu.application.R
import edu.application.databinding.FragmentSettingsColorsSchemeBinding
import edu.application.ui.fragments.navigation.adapters.TextFormatter.setBorder
import edu.application.ui.fragments.navigation.adapters.TextFormatter.setColor
import edu.application.ui.fragments.navigation.adapters.TextFormatter.sharedPreferences

class ColorSchemeFragment : Fragment() {

    private lateinit var binding: FragmentSettingsColorsSchemeBinding
    private lateinit var editor: SharedPreferences.Editor

    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?,
                              savedInstanceState: Bundle?): View {
        binding = FragmentSettingsColorsSchemeBinding.inflate(inflater, container, false)
        editor = sharedPreferences.edit()

        setColor(binding.customBtn)
        setBorder(binding.customBtn)

        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding.lightBtn.setOnClickListener {
            editor.remove("lesson_background_color")
            editor.remove("lesson_text_color")
            editor.apply()
            findNavController(requireActivity(), R.id.nav_host_fragment)
                .navigate(R.id.action_colorSchemeFragment_to_colorSchemeFragment)
        }

        binding.customBtn.setOnClickListener {
            findNavController(requireActivity(), R.id.nav_host_fragment)
                .navigate(R.id.action_colorSchemeFragment_to_colorSchemeCustomFragment) }
    }
}